modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/util/fieldSupport.es.js

```diff
@@ -260,14 +260,21 @@ export const createField = (props, event) => {
 		},
 	};
 
-	const {fieldName, fieldReference, name, settingsContext} = newField;
+	const {
+		editorConfig,
+		fieldName,
+		fieldReference,
+		name,
+		settingsContext,
+	} = newField;
 
 	return {
 		...getFieldProperties(
 			settingsContext,
 			defaultLanguageId,
 			editingLanguageId
 		),
+		editorConfig,
```

Comment:
I understand what you're doing here and it works like a charm. 

My only concern is that we are referencing `editorConfig` in a general purpose file that I believe is used for all the other fields that don't need to know about the `editorConfig`. I would have expected this to be just inside some `RichText` file.  Same thing for the Java code.

But don't worry too much because I'm new to this codebase and I am just beginning to grasp it.

Reply:
@marcosapmf I have some ideas on how to improve it. I will give it a try and maybe send a follow up.

Reply:
Hey @4lejandrito, I see, but in this case, I'm afraid we don't have a specific file for taking care the RichText stuff when we are adding a field. The same thing for the Java code, but if we need to do that for other fields we can think on a more generic solution 🙂
What do you think?

Reply:
I think you're right. It is too soon to expose a new Java API for this without having other use cases. Then I would say this is ready to forward.